cmd/query/main.go

```diff
@@ -83,13 +83,12 @@ func main() {
 				logger.Fatal("Cannot create metrics factory.", zap.Error(err))
 			}
 
-			tracer, closer, err := jaegerClientConfig.Configuration{
-				Sampler: &jaegerClientConfig.SamplerConfig{
-					Type:  "probabilistic",
-					Param: 1.0,
-				},
-				RPCMetrics: true,
-			}.New(
+			cfg, err := jaegerClientConfig.FromEnv()
```

Comment:
I think this would effectively disable tracing in the query service since it's typically a low QPS service and the default 0.001 probability will kick in. I would rather have a default configuration like above that can be overwritten via env variables.

Reply:
Sure. That really makes sense. So, I shall check if the above fields are nil and if so will set the default configuration.

Reply:
Does this apply for both the Sampler and RPCMetrics?

Reply:
I don't think doing it here is the best solution. I would rather allow Configuration class to be created with default values, and then allow something like InitFromEnv() which will take care of overriding configuration parameters as needed. cc @jpkrohling

Reply:
Agree with @yurishkuro: ideally, we should keep the same behavior for the end user as we have now, just with the added possibility of customizing the behavior when and where needed.

Reply:
Thanks for the comment @jpkrohling . If my understanding is correct, within the FromEnv(), there should be IntFromEnv() which gets the default configuration set that is then overridden by the environmental variables.
May I implement this flow?

Reply:
@yurishkuro do you mean something like this?

```golang
func InitFromEnv(Configuration defaults) (*Configuration, error) {
...
}
```

Reply:
my proposal requires changes to the Jaeger Go client

Reply:
@subashiniganesh I also got a bit confused by @yurishkuro's comment, but it doesn't make sense (to me) to have `NewDefaultConfiguration()` in the Jaeger Go Client. So, the changes required are:

1. Add `func NewDefaultConfiguration() Configuration {}` to an appropriate place. I would suggest `pkg/config`, but take a look and see if there's a better place for it. This code would basically do what the [current `cmd/query/main.go` is doing](https://github.com/jaegertracing/jaeger/blob/master/cmd/query/main.go#L86-L96) regarding the client config.
1. Add `func (c *Configuration) InitFromEnv() error {}` to the [Jaeger Go Client](https://github.com/jaegertracing/jaeger-client-go), under the `config` package, which will override the config options from `c` with the ones from the env vars[1]. Take a look at the `config_env.go` source file: you can probably reuse a lot of that, avoiding duplication.
1. Change the `cmd/query/main.go` like you did in this PR, but calling `config.NewDefaultConfiguration().InitFromEnv()` instead of `jaegerClientConfig.FromEnv()`

Reply:
NewDefaultConfig function, hopefully, should not be necessary since the empty struct should work as the default config.

Reply:
> the empty struct should work as the default config.

This is what's confusing me :) The default config for the query component is not the same as the default config for the Go client (the query config sets the sampling to const 1).

Reply:
Then three steps, default config from client (just a struct), override sampling, then override from env. The first two steps are already done in query svc, just need the client change.